### ISSUE TITLE
react-hooks/exhaustive-deps treats imports and module-level destructuring as reactive dependencies

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1230]
## Summary

Imported bindings and bindings destructured at module scope are reported as missing dependencies. They are outside the component render scope; ESLint does not require them. A simple module-level const object is handled correctly.

Verified with @utoo/lint 0.4.1, ESLint 8.57.1, @typescript-eslint/parser 6.21.0, eslint-plugin-react-hooks 4.6.2, Node.js 22.23.1 on macOS arm64.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.1 eslint@8.57.1 @typescript-eslint/parser@6.21.0 eslint-plugin-react-hooks@4.6.2
```

`utlint.config.json`:

```json
{
  "rules": {
    "react-hooks/exhaustive-deps": "warn"
  }
}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "react-hooks"
  ],
  "rules": {
    "react-hooks/exhaustive-deps": "warn"
  }
};
```

Run each example separately using:

```sh
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
```

## Reproduction 1

`example.tsx`:

```tsx
import { useEffect } from 'react'; import { log } from './helpers';
export function Example(){ useEffect(()=>log(),[]);return null; }
```

`helpers.ts`:

```ts
export function log(){}
```

```text
ESLint:
(no diagnostics)

utlint:
React Hook useEffect has a missing dependency: 'log'. Either include it or remove the dependency array.
```

## Reproduction 2

`example.tsx`:

```tsx
import { useMemo } from 'react';const {value} = window.settings;
export function Example(){return useMemo(()=>value,[]);}
```

```text
ESLint:
(no diagnostics)

utlint:
React Hook useMemo has a missing dependency: 'value'. Either include it or remove the dependency array.
```

## Controls

`example.tsx`:

```tsx
import { useMemo } from 'react';const options={name:'hello'};
export function Example(){ const result=useMemo(()=>options.name,[]);return result; }
```

```text
ESLint:
(no diagnostics)

utlint:
(no diagnostics)
```

All examples parse successfully. No lint-disable comments or additional rules are involved.
